src/features/offer/services/useCtaWording.ts

```diff
@@ -0,0 +1,20 @@
+import { t } from '@lingui/macro'
+
+import { CategoryType } from 'api/gen'
+import { useAuthContext } from 'features/auth/AuthContext'
+import { useUserProfileInfo } from 'features/home/api'
+import { _ } from 'libs/i18n'
+
+interface CtaWordingI {
+  categoryType: CategoryType
+}
+
+export const useCtaWording = ({ categoryType }: CtaWordingI) => {
+  const { isLoggedIn } = useAuthContext()
+  const { data: profileInfo } = useUserProfileInfo()
+  if (!isLoggedIn || (profileInfo && !profileInfo.isBeneficiary))
```

Comment:
Pour expliquer la condition :
```suggestion
  const isNotBeneficiaryUser = !isLoggedIn || (profileInfo && !profileInfo.isBeneficiary)
  if (isNotBeneficiaryUser) {
    ...
   }
```
Encore mieux, sans condition négative :
```suggestion
  const isBeneficiaryUser = isLoggedIn && profileInfo && profileInfo.isBeneficiary
  if (isBeneficiaryUser) {
    return _(t`Voir les disponibilités`)
   }
   ...
   
```
Le Why vient de Clean Code : nommer les conditions avec des variables pour les expliquer, et dans la mesure du possible ne pas utiliser de condition négative pour faciliter la lecture.

Reply:
En effet c'est plus clair ainsi
Même si ça reste clair sans.

J'ai déployé avant de voir cela, ce sera une bonne refacto pour les prochains tickets (PC-6003) par exemple